homeassistant/components/onvif/config_flow.py

```diff
@@ -205,10 +205,23 @@ async def async_step_profiles(self, user_input=None):
 
             # Get the MAC address to use as the unique ID for the config flow
             if not self.device_id:
-                network_interfaces = await device_mgmt.GetNetworkInterfaces()
-                for interface in network_interfaces:
-                    if interface.Enabled:
-                        self.device_id = interface.Info.HwAddress
+                try:
+                    network_interfaces = await device_mgmt.GetNetworkInterfaces()
+                    for interface in network_interfaces:
+                        if interface.Enabled:
+                            self.device_id = interface.Info.HwAddress
+                except Fault as fault:
+                    if (
+                        "'ns0:GetNetworkInterfaces' not implemented"
```

Comment:
I'd simplify this to just `"not implemented"` since the namespace isn't guaranteed to be `ns0`.